### ISSUE TITLE
Linux snap package O3DE installer: Editor fails to launch due to assert and error on Ubuntu 20.04

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#15939]
**Describe the bug**
Launching Editor from Project Manager on the O3DE snap package (`23.05.0`) results in the following errors and an assert being printed in the Terminal:
```
Module: Attempting to load module:/snap/o3de/x1/23.05.0/bin/Linux/profile/Default/libEditorLib.so
Module: Failed with error:
/snap/core22/current/lib/x86_64-linux-gnu/libgssapi_krb5.so.2: undefined symbol: k5_externalize_context, version krb5_3_MIT
```
Assert Information:

**File:** 
`/data/workspace/o3de/Code/Editor/main.cpp`
**Condition:** 
```
(140022714637568) 'int main(int, char **)'
System: EditorLib could not be loaded
System:  -- error: unable to obtain symbol name for this frame
```
**Line:** 26

Attempting to launch the O3DE with `sudo` did not help. 
Please note that the issue **does not** occur for https://o3debinaries.org/2.0.0/Linux/o3de_2.0.0_amd64.snap snap package, however, for that package the https://github.com/o3de/o3de/issues/15856 occurs. 
The issue does not reproduce on Ubuntu 22.04. Please refer to the attached video and Terminal log for more information.

**Steps to reproduce**
1. Download the O3DE snap package from https://o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap.
2. Follow instructions listed in https://www.o3de.org/docs/welcome-guide/setup/installing-linux/#installing-o3de-from-a-snap-package.
3. Launch Project Manager as mentioned in the instruction in step 2.
4. Create a project.
5. Build the created project.
6. Press "Open Editor" button after the project is built.

**Expected behavior**
Editor launches without issues.

**Actual behavior**
Editor fails to launch any project.

**Screenshots/Video**

https://github.com/o3de/o3de/assets/86953659/fefa3b8d-4bf6-4656-8eb0-097058550cef

**Found in Branch**
https://o3debinaries.org/23.05.0/Linux/o3de_23.05.0_amd64.snap (downloaded 5/12/2023 at 10:30 AM GMT+2)

**Desktop/Device**
- Device: PC
- OS: Linux
- Version: Ubuntu 20.04.6 LTS
- CPU: CPU: Intel Core i7-11700F
- GPU: NVIDIA GeForce RTX 3070
- Memory: 32GB

**Additional context**
[Open_Editor_Fail.txt](https://github.com/o3de/o3de/files/11462956/Open_Editor_Fail.txt)

**Comment by spham-amzn** [comment]
> Can you run the following command:
> 
> ```
> ldd /snap/o3de/x1/23.05.0/bin/Linux/profile/Default/libEditorLib.so
> ```
> 
> and report what the output is?

Disregard this request, we checked and it looks like an issue with the package libraries that are in the snap container that link outside to the system libc.so library. The modules were built linking against a 22.04 version of libc.so, so linking to an older one failed because it is libc.so is not forward compatible.

**Comment by byrcolin** [comment]
So I think the answer here is to fix/remake the 20.04 package? @spham-amzn @amzn-changml

**Comment by amzn-changml** [comment]
The snap package for 2.1.0 has been updated with references to core20. @LB-MateuszSzweda please redownload and try it again

**Comment by LB-MichalRodzos** [comment]
I verified the .snap installer 2.1.0 today (downloaded  6/02/2023 between 8 and 9AM GMT+2). I was able to create and built a project, and then launch the Editor.

**Comment by LB-MateuszSzweda** [comment]
@amzn-changml I confirm that I was also able to build and launch projects with .snap installer 2.1.0 today (downloaded 6/02/2023 between 15 and 16PM GMT+2) on my Ubuntu 20.04 machine.

**Comment by amzn-changml** [comment]
Thanks! I'll go ahead and close this issue

**Comment by LB-KacperKapusta** [comment]
Issue verified as fixed using the latest 2.1.0 .snap package.
